src/plugins/index_pattern_management/server/routes/resolve_index.ts

```diff
@@ -63,13 +63,25 @@ export function registerResolveIndexRoute(router: IRouter): void {
         ? context.dataSource.opensearch.legacy.getClient(dataSourceId).callAPI
         : context.core.opensearch.legacy.client.callAsCurrentUser;
 
-      const result = await caller('transport.request', {
-        method: 'GET',
-        path: `/_resolve/index/${encodeURIComponent(req.params.query)}${
-          queryString ? '?' + new URLSearchParams(queryString).toString() : ''
-        }`,
-      });
-      return res.ok({ body: result });
+      try {
+        const result = await caller('transport.request', {
+          method: 'GET',
+          path: `/_resolve/index/${encodeURIComponent(req.params.query)}${
+            queryString ? '?' + new URLSearchParams(queryString).toString() : ''
+          }`,
+        });
+        return res.ok({ body: result });
+      } catch (err) {
+        return res.customError({
+          statusCode: err.statusCode || 500,
```

Comment:
I thought the redirect happens during this API call, but we are not handling the error using the `createDataSourceError` here?

Reply:
401 is causing refresh by security plugin. Prior to here, we use `createDataSourceError()` to cast the 401 to 400, here it receives that 400, and will never get any 401 and won't trigger any refresh

Reply:
I missed it, it is thrown during constructing the client